cli/cmd/plugin/unmanaged-cluster/cluster/minikube.go

```diff
@@ -0,0 +1,122 @@
+// Copyright 2022 VMware Tanzu Community Edition contributors. All Rights Reserved.
+// SPDX-License-Identifier: Apache-2.0
+
+package cluster
+
+import (
+	"fmt"
+	"os"
+	"os/exec"
+	"path/filepath"
+
+	"github.com/vmware-tanzu/community-edition/cli/cmd/plugin/unmanaged-cluster/config"
+)
+
+var minikubeBinName = "minikube"
+
+type MinikubeClusterManager struct{}
+
+// Create
+func (mkcm MinikubeClusterManager) Create(c *config.UnmanagedClusterConfig) (*KubernetesCluster, error) {
+	if !mkcm.isMinikubeRunning() {
+		// TODO: Is minikube it the current context? If not, attempt to switch to it
+
+		args := []string{
+			"start",
+		}
+
+		cmd := exec.Command(minikubeBinName, args...)
+
+		err := cmd.Run()
+		if err != nil {
+			return nil, err
+		}
+	}
+
+	// readkubeconfig in bytes from configured kubeconfig.
```

Comment:
I'm a little worried about this behavior and taking over an existing `minikube` cluster we find in the user's `kubeconfig`.

Instead, I'd like us to leverage `minikube`'s `profile` feature, which enables having multiple clusters on our machine:

https://minikube.sigs.k8s.io/docs/faq/#how-can-i-create-more-than-one-cluster-with-minikube

So, in summary, we never take over an existing cluster. If users want to do that, they should use our `-e` flag to point to the already running cluster.

When we create clusters, we should leverage the profile specification, for example:

```
minikube start -p p1 --driver=docker 
```

The value of `-p` should be the cluster name. We should decide on a default driver, but also let our users specify it through our providerConfiguration.

On cluster create, we query for the existing profiles:

```
minikube profile list -o json | jq .valid[].Name

"minikube"
```

If a name in the profile matches what the user set as the **cluster name**, we should fail our create.